transaction_metadata/index.js

```diff
@@ -0,0 +1,245 @@
+const fs = require("fs");
```

Comment:
I've extensively reviewed and tested this code. Obviously it works and the final binary passes the applicable tests. I'll grant that the Merkle tree implementation is elegant, however, I'm concerned about whether it is the most efficient and simple approach, as I would not want to add unneeded complexity. Is a Merkle tree better for this purpose than a list of hashes?

Reply:
Thepoint of merkle trees is to save space in the client app. The current approach, before the introduction of merkle trees, needs approximately 200b per tx (metadata stored in the app). This leads to